src/Dapr.Actors/Runtime/ActorHost.cs

```diff
@@ -160,7 +164,10 @@ public static ActorHost CreateForTest(Type actorType, string actorTypeName, Acto
         /// Gets the <see cref="ActorTimerManager" />.
         /// </summary>
         public ActorTimerManager TimerManager { get; }
-
-        internal DaprStateProvider StateProvider { get; set; }
+        
+        /// <summary>
+        /// Gets the <see cref="DaprStateProvider" />.
+        /// </summary>
+        internal IDaprStateProvider StateProvider { get; }
```

Comment:
Technically, changing this to an interface is actually a breaking change, so we can't accept it. I think you can accomplish this without it though.